lib/parser.ex

```diff
@@ -8,76 +8,69 @@ defmodule Membrane.Ogg.Parser do
 
     @type t() :: %__MODULE__{
             payload: binary(),
-            track_id: non_neg_integer(),
             bos?: boolean(),
             eos?: boolean()
           }
 
     defstruct payload: <<>>,
-              track_id: 0,
               bos?: false,
               eos?: false
   end
 
-  @type continued_packets_t :: %{integer => binary}
-
-  @spec parse(binary, continued_packets_t) ::
-          {parsed :: [Packet.t()], continued_packets :: continued_packets_t, rest :: binary}
-  def parse(data, continued_packets) do
-    do_parse([], data, continued_packets)
+  @spec parse(binary(), binary() | nil) ::
+          {parsed :: [Packet.t()], new_continued_packet :: binary(), rest :: binary()}
+  def parse(data, continued_packet) do
+    do_parse([], data, continued_packet)
   end
 
-  defp do_parse(acc, data, continued_packets) do
-    case parse_page(data, continued_packets) do
+  defp do_parse(acc, data, continued_packet) do
+    case parse_page(data, continued_packet) do
       {:error, :need_more_bytes} ->
-        {acc, continued_packets, data}
+        {acc, continued_packet, data}
 
       {:error, :invalid_crc} ->
         raise "Corrupted stream: invalid crc"
 
       {:error, :invalid_header} ->
         raise "Corrupted stream: invalid page header"
 
-      {:ok, packets, continued_packets, rest} ->
-        do_parse(acc ++ packets, rest, continued_packets)
+      {:ok, packets, incomplete_packet, rest} ->
+        do_parse(acc ++ packets, rest, incomplete_packet)
     end
   end
 
-  defp parse_page(initial_bytes, continued_packets) do
-    with {:ok, data, header_type, bitstream_serial_number, number_of_page_segments} <-
-           parse_header(initial_bytes),
+  defp parse_page(initial_bytes, continued_packet) do
+    with {:ok, data, header_type, number_of_page_segments} <- parse_header(initial_bytes),
          {:ok, segment_table, data} <- parse_segment_table(data, number_of_page_segments),
          :ok <- verify_length_and_crc(initial_bytes, segment_table) do
-      {packets, page_continued_packet, rest} = parse_segments(data, segment_table)
+      {packets, incomplete_packet, rest} = parse_segments(data, segment_table)
 
-      {packets, continued_packets} =
+      {packets, new_incomplete_packet} =
```

Comment:
I'd avoid creating a `new_` variable unless you need the 'old' variable later, since it's easy to use the 'old' variable accidentally
```suggestion
      {packets, incomplete_packet} =
```